heareval/predictions/task_predictions.py

```diff
@@ -845,7 +885,7 @@ def task_predictions_train(
         logger=logger,
     )
     train_dataloader = dataloader_from_split_name(
-        "train",
+        data_splits["train"],
```

Comment:
Again this stuff is so confusing. Can we put all this weird logic in one place with a clear explanation, so this weird pattern doesn't occur throughout this file? The fewer patterns you have to remember, the better

Reply:
# 319